---
title: "limony"
subtitle: "lake itag measurements over nineteen years"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{introduction}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

Please cite: _Data Announcement citation_

## get started

This package includes the limony dataset and functions to subset it. The limony dataset is a nineteen year 16S rRNA gene amplicon dataset from Lake Mendota, WI USA. More information about the dataset itself is available in the data announcement: _citation_

```{r setup}
# Load the limony package
library(limony)
# Load the limony data
data("limony")
```

Let's explore the nested list data structure:  
```{r eval=TRUE}
# The top level lists contain names, abundances, etc
names(limony)

# The nested lists are for each taxon level
names(limony$av)
```
list name   | Matrix contents
------------|-----------------------
names       | Taxonomy Assignments (character)
av          | Average abundances (numeric, the OTU table)
sd          | Standard deviations (numeric)
bq          | Below quantification (logical, true if pd OR br)
pd          | Partial detection (logical, true if not present in all replicates)
br          | Below resolution (logical, true if abundance is lower than a single read at the worst sequencing depth)

The limony package lets you easily subset this list structure to extract the dates or taxa you are interested in, while maintaining all the other information about standard deviation and quantification limits. This vignette will go through how to use the subsetting functions. 

## get OTU table
_"But wait! I just want an OTU table!"_ No problem, do this and skip the rest of the vignette :)
```{r eval=TRUE}
otu.table <- limony$av$seqID
otu.taxonomy <- limony$names$seqID
sample.dates <- convert.sample.names.to.dates(sample.names = colnames(otu.table))
otu.table[1:3,1:3] # numeric matrix
otu.taxonomy[1:3,1:3] # character matrix
sample.dates[1:3] # dates vector
```

## subset by sample 

First get the column indexes of the samples using `get.sample.inexes` and then subset limony using `subset.by.sample`.

Example: Remove non-standard samples that were pre-filtered:
```{r, eval=FALSE}
index <- get.sample.indexes(my.list = limony, remove.prefiltered = TRUE)
no.pf <- subset.by.sample(my.list = limony, keep.index = index)
```

Example: Keep only samples from 2015:
```{r, eval=FALSE}
index <- get.sample.indexes(my.list = limony, start.YY.MM.DD = "15-1-1", end.YY.MM.DD = "15-12-31")
only.2015 <- subset.by.sample(my.list = limony, keep.index = index)
```

Example: Keep only samples from June, but across all years:
```{r, eval=FALSE}
index <- get.sample.indexes(my.list = limony, start.YY.MM.DD = "00-6-1", end.YY.MM.DD = "00-6-31", dates.are.season.range = TRUE) # year is ignored, can be any year
only.June <- subset.by.sample(my.list = limony, keep.index = index)
```

## subset by taxa

First you flatten the list structure to a single taxon level using `flatten.to.single.level`, then you find the indexes of the taxa you want to keep using `get.taxon.indexes`, then you subset to those taxa using `subset.by.taxa`, and finally you re-group to get all the taxonomy levels again using `group.by.tax`. By re-grouping the taxonomy lists you can keep track of the average and standard deviation at each taxonomy level.

Example: Keep only acI
```{r eval=FALSE}
flat.list <- flatten.to.single.level(my.list = limony)
index <- get.taxon.indexes(my.list = flat.list, taxa = "acI")
flat.acI <- subset.by.taxa(my.list = flat.list, keep.index = index)
acI <- group.by.tax(my.list = flat.acI)
````

Example: Remove chloroplasts and renormalize the data
```{r eval=FALSE}
flat.list <- flatten.to.single.level(my.list = limony)
index <- get.taxon.indexes(my.list = flat.list, taxa = "Chloroplast")
flat.no.chloro <- subset.by.taxa(my.list = flat.list, keep.index = -index) # note the minus sign in front of index
no.chloro <- group.by.tax(my.list = flat.no.chloro)
```

## spell the taxa names correctly
Subsetting by taxa can be annoying, because the taxonomy name is case sensitive and must be spelled correctly. To help you figure out what the correct names are, there are is a  handy function called `print.names.under.taxon`.

Example: I want to subset Microcystis, but I don't know exactly what it's called in Silva.
```{r eval=TRUE}
# I know it's a "Cyanobacteriales"
print.names.under.taxon(my.list = limony, taxon = "Cyanobacteriales")
# Aha! Looks like it's "Microcystaceae".
```
Example putting everything together:
```{r eval=TRUE}
# Now I can use this spelling to subset out Microcystis in July:
flat.lim <- flatten.to.single.level(my.list = limony)

index <- get.taxon.indexes(my.list = flat.lim, taxa = "Microcystaceae")

microcystis <- subset.by.taxa(my.list = flat.lim, keep.index = index)

index <- get.sample.indexes(my.list = microcystis, start.YY.MM.DD = "00-7-1", end.YY.MM.DD = "19-7-31", dates.are.season.range = TRUE, remove.prefiltered = TRUE)

microcystis <- subset.by.sample(my.list = microcystis, keep.index = index)

microcystis <- group.by.tax(my.list = microcystis)

# And I can look at Microcystis OTUs:
microcystis$av$seqID[1:3,1:3]

# Or at the Microcystis Genus
microcystis$av$`Genus/Clade`[ ,1:3, drop=FALSE]
```

## contact me

If you have more questions about the limony dataset or the limony package, or if you want to discuss a planned use of this data, please contact me: Robin.Rohwer@gmail.com  